pubsub/azuresb/azuresb.go

```diff
@@ -506,16 +525,32 @@ func (s *subscription) updateMessageDispositions(ctx context.Context, ids []driv
 		return nil
 	}
 
-	lockIds := []amqp.UUID{}
-	for _, mid := range ids {
-		if id, ok := mid.(*uuid.UUID); ok {
-			lockTokenBytes := [16]byte(*id)
-			lockIds = append(lockIds, amqp.UUID(lockTokenBytes))
+	// Group by partitionID; AzureSB doesn't support updating dispositions for
+	// messages from different partitions.
+	partitions := map[int16][]amqp.UUID{}
+	for _, ackID := range ids {
+		if pid, ok := ackID.(*partitionAckID); ok {
+			lockTokenBytes := [16]byte(*pid.LockToken)
+			partitions[pid.PartitionID] = append(partitions[pid.PartitionID], amqp.UUID(lockTokenBytes))
 		}
 	}
+
+	// Update all partitions in parallel.
+	g, ctx := errgroup.WithContext(ctx)
+	for _, lockTokens := range partitions {
+		g.Go(func() error {
```

Comment:
Do we ever expect the number of partitions to be large? The parallelism here is unbounded.

(Also, does this end up being an issue with HTTP replay due to nondeterminism?)

Reply:
I doubt it, but better safe than sorry. Fixed to max out at 5 goroutines.

There's no HTTP replay for `azuresb` :-(.